external/src/main/java/com/redhat/cloud/custompolicies/engine/actions/plugins/BetaHooksActionPluginRegister.java

```diff
@@ -0,0 +1,108 @@
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.Dependent;
+import javax.inject.Inject;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+@Plugin(name = "hooks")
+@Dependent
+public class BetaHooksActionPluginRegister implements ActionPluginListener {
+    private final MsgLogger log = MsgLogging.getMsgLogger(BetaHooksActionPluginRegister.class);
+
+    private static String APPLICATION_NAME = "custom-policies";
+    private static String EVENT_TYPE = "custom-policies";
+    private static String LEVEL = "1";
+
+    @Inject
+    Vertx vertx;
+
+    @ConfigProperty(name="external.notifications-backend.server")
+    String notificationsHostname;
+
+    @ConfigProperty(name="external.notifications-backend.port")
+    Integer notificationsPort;
+
+    @Inject
+    @Channel("hooks")
+    Emitter<JsonObject> channel;
+
+    private WebClient client;
+
+    @PostConstruct
+    public void initialize() {
+        client = WebClient.create(vertx, new WebClientOptions().setDefaultHost(notificationsHostname).setDefaultPort(notificationsPort));
+        client.post(String.format("/apps"))
+                .sendJsonObject(createApplicationRegistrationPayload())
+                .thenApply(resp -> {
+                    if(resp.statusCode() == 200) {
+                        log.info("Application registered to beta hooks backend");
+                    } else {
+                        log.error("Failed to register application to beta hooks backend: " + resp.bodyAsString());
+                    }
+                    return null;
+                });
+    }
+
+    private JsonObject createApplicationRegistrationPayload() {
+        JsonObject registration = new JsonObject();
+        JsonObject application = new JsonObject();
+        application.put("name", APPLICATION_NAME);
+        application.put("title", "Custom Policies");
+        registration.put("application", application);
+
+        JsonObject eventTypes = new JsonObject();
+        eventTypes.put("id", "any");
+        eventTypes.put("title", "Any policy");
+        eventTypes.put("levels", new JsonObject());
+
+        registration.put("event_types", eventTypes);
+        return registration;
+    }
+
+    @Override
+    public Set<String> getProperties() {
+        return new HashSet<>();
+    }
+
+    @Override
+    public Map<String, String> getDefaultProperties() {
+        Map<String, String> defaultProperties = new HashMap<>();
+        defaultProperties.put("_managed", "true");
+        return defaultProperties;
+    }
+
+    @Override
+    public void process(ActionMessage msg) throws Exception {
+        // Fields and terminology straight from the target project
+        JsonObject message = new JsonObject();
+        message.put("application", APPLICATION_NAME);
+        message.put("event_type", EVENT_TYPE);
+        message.put("level", 1); // Is this necessary field?
```

Comment:
I don't think it can be "1" because it needs to match a "level" defined during registration.
I asked on Slack it it should be empty or just not there at all:
https://ansible.slack.com/archives/CFKVBJ6M6/p1582213163012600

Reply:
So looks like the value should be null according to @adamruzicka 
https://ansible.slack.com/archives/CFKVBJ6M6/p1582281814015800